app/services/utils.ts

```diff
@@ -80,4 +80,18 @@ export default class Utils {
     }
     return result;
   }
+
+  static getAvailableWhitelistedType  (obsAvailableTypes: string[], whitelistedTypes:  any[]):  any[] {
```

Comment:
I don't think you really need to define a function here.  You can replicate this same behavior using this line of code:
```TypeScript
whitelistedTypes.filter(type => obsAvailableTypes.includes(type.value));
```